Frends.ServiceBus.Read/Frends.ServiceBus.Read/Definitions/MessagingFactory.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Microsoft.Azure.ServiceBus;
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Frends.ServiceBus.Read.Definitions;
+
+/// <summary>
+/// Class handles clients for the service bus. Enables cached connections to the service bus.
+/// </summary>
+public sealed class ServiceBusMessagingFactory : IDisposable
+{
+    private static readonly Lazy<ServiceBusMessagingFactory> instanceHolder = new(() => new ServiceBusMessagingFactory());
+    /// <summary>
+    /// The ServiceBusMessagingFactory singleton instance
+    /// </summary>
+    public static ServiceBusMessagingFactory Instance
+    {
+        get { return instanceHolder.Value; }
+    }
+
+    private static readonly object factoryLock = new();
+    private readonly ConcurrentDictionary<string, ServiceBusConnection> _connections = new();
+
+    private ServiceBusMessagingFactory()
+    {
+
+    }
+
+
+    /// <summary>
+    /// Create message receiver for the given connection string and entity path
+    /// </summary>
+    /// <param name="connectionString">Connection string</param>
+    /// <param name="path">Name of the queue</param>
+    /// <param name="timeout">TimeoutSeconds</param>
+    /// <returns></returns>
+    public MessageReceiver GetMessageReceiver(string connectionString, string path, TimeSpan timeout)
+    {
+        var receiver = new MessageReceiver(GetCachedMessagingFactory(connectionString, timeout), path, receiveMode: ReceiveMode.ReceiveAndDelete);
+
+        return receiver;
+    }
+
+    /// <summary>
+    /// Create a message sender for the given connection string and entity path
+    /// </summary>
+    /// <param name="connectionString"></param>
+    /// <param name="path"></param>
+    /// <param name="timeout"></param>
+    /// <returns></returns>
+    public MessageSender GetMessageSender(string connectionString, string path, TimeSpan timeout)
+    {
+        return new MessageSender(GetCachedMessagingFactory(connectionString, timeout), path);
+    }
+
+    private ServiceBusConnection GetCachedMessagingFactory(string connectionString, TimeSpan timeout)
+    {
+        string key = timeout.TotalSeconds + "-" + connectionString;
+
+        if (!_connections.ContainsKey(key))
+        {
+            lock (factoryLock) // TODO: change double check
```

Comment:
Why this is still todo?

Reply:
MessagingFactory.cs has been unchanged from Frends.ServiceBus to Frends.AzureServiceBus tasks and, while it sure need some update, I don't think it's a good idea to modify it for the first net6 release.